enginetest/server_engine.go

```diff
@@ -72,74 +78,139 @@ func NewServerQueryEngine(t *testing.T, engine *sqle.Engine, builder server.Sess
 		t:      t,
 		engine: engine,
 		server: s,
+		port:   p,
 	}, nil
 }
 
-func newConnection(ctx *sql.Context) (*gosql.DB, error) {
+// NewConnection creates a new connection to the server regardless of whether there is an existing connection.
+// If there is an existing connection, it closes it and creates a new connection. This method allows running
+// multiple queries in the same session. Otherwise, new connection uses new session that some session state data
+// will not be persisted. This function is also called when there is no connection is set while running a query.
+func (s *ServerQueryEngine) NewConnection(ctx *sql.Context) error {
+	if s.conn != nil {
+		err := s.conn.Close()
+		if err != nil {
+			return err
+		}
+	}
+
 	db := ctx.GetCurrentDatabase()
 	// https://stackoverflow.com/questions/29341590/how-to-parse-time-from-database/29343013#29343013
-	return gosql.Open("mysql", fmt.Sprintf("root:@tcp(127.0.0.1)/%s?parseTime=true", db))
+	conn, err := gosql.Open("mysql", fmt.Sprintf("root:@tcp(127.0.0.1:%d)/%s?parseTime=true", s.port, db))
+	if err != nil {
+		return err
+	}
+	s.conn = conn
+	return nil
 }
 
-func (s ServerQueryEngine) PrepareQuery(ctx *sql.Context, query string) (sql.Node, error) {
+func (s *ServerQueryEngine) PrepareQuery(ctx *sql.Context, query string) (sql.Node, error) {
+	if s.conn == nil {
+		err := s.NewConnection(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
 	// TODO
 	// q, bindVars, err := injectBindVarsAndPrepare(s.t, ctx, s.engine, query)
```

Comment:
Is there a reason we don't turn this on yet? I see we're preparing the query in the `Query` method, so just curious if we're ready to turn that some code on here.